Chargebee/Classes/Purchase/CBPurchaseManager.swift

```diff
@@ -153,6 +163,29 @@ public extension CBPurchase {
             }
         }
     }
+    
+    func purchaseProductHandlerForNonSubscriptions(product: CBProduct,completion handler: @escaping ((_ result: Result<(NonSubscription), Error>) -> Void)) {
+        
+        guard CBAuthenticationManager.isSDKKeyPresent() else {
+            handler(.failure(CBPurchaseError.cannotMakePayments))
+            return
+        }
+
+        if !CBPurchase.shared.canMakePayments() {
+            handler(.failure(CBPurchaseError.cannotMakePayments))
+        } else {
+            authenticationManager.isSDKKeyValid { status in
+                if status {
+                    let payment = SKPayment(product: product.product)
+                    SKPaymentQueue.default().add(payment)
+
+                } else {
+                    handler(.failure(CBPurchaseError.invalidSDKKey))
+                }
+            }
+        }
+    }
```

Comment:
This logic looks exactly the same as that of the `purchaseProductHandler`. Can we reuse it instead of duplicating?

Reply:
Made it generic function because completion handler is different for both and i made it reusable ,updated